src/js/base/module/AutoReplace.js

```diff
@@ -0,0 +1,88 @@
+import lists from '../core/lists';
+import dom from '../core/dom';
+import key from '../core/key';
+
+export default class AutoReplace {
+  constructor(context) {
+    this.context = context;
+    this.options = context.options.replace || {};
+
+    this.keys = [key.code.ENTER, key.code.SPACE, key.code.PERIOD, key.code.COMMA, key.code.SEMICOLON, key.code.SLASH];
+    this.previousKeydownCode = null;
+
+    this.events = {
+      'summernote.keyup': (we, e) => {
+        if (!e.isDefaultPrevented()) {
+          this.handleKeyup(e);
+        }
+      },
+      'summernote.keydown': (we, e) => {
+        this.handleKeydown(e);
+      }
+    };
+  }
+
+  shouldInitialize() {
+    return !!this.options.match;
+  }
+
+  initialize() {
+    this.lastWord = null;
+  }
+
+  destroy() {
+    this.lastWord = null;
+  }
+
+  /**
+   * options.replace.match = function(word, done){ if(word=='this') done('that') else done();
```

Comment:
Is this for an example?

Reply:
no, it is my mistake to test auto replace module.   
There is must removed.